api/tests/test_api.py

```diff
@@ -190,3 +193,34 @@ def test_get(self):
     def test_follower_require_login(self):
         res = self.client.get(f'/api/v1/authors/{self.author.id}/followers/')
         self.assertEqual(res.status_code, 403)
+
+    def test_add_follower(self):
+        self.client.login(username='bob', password='password')
+        res = self.client.put(f'/api/v1/authors/{self.author.id}/followers/{self.other_user3.id}/')
+        self.assertEqual(len(Follow.objects.filter(followee=self.author)), 3)
+        self.assertEqual(res.status_code, 200)
+
+    def test_add_follower_duplicate(self):
+        self.client.login(username='bob', password='password')
+        res = self.client.put(f'/api/v1/authors/{self.author.id}/followers/{self.other_user2.id}/')
+        self.assertEqual(len(Follow.objects.filter(followee=self.author)), 2)
+        # this need to be verify later
+        self.assertEqual(res.status_code, 200)
+
+    def test_add_follower_not_exit(self):
+        self.client.login(username='bob', password='password')
+        res = self.client.put(f'/api/v1/authors/{self.author.id}/followers/100/')
+        self.assertEqual(len(Follow.objects.filter(followee=self.author)), 2)
+        self.assertEqual(res.status_code, 404)
+
+    def test_delete_follower(self):
+        self.client.login(username='bob', password='password')
+        res = self.client.delete(f'/api/v1/authors/{self.author.id}/followers/{self.other_user2.id}/')
```

Comment:
Wondering if this test case would be a bit stronger if we verified there were 2 before, and 1 after the `DELETE` call:
```suggestion
        self.assertEqual(len(Follow.objects.filter(followee=self.author)), 2)
        res = self.client.delete(f'/api/v1/authors/{self.author.id}/followers/{self.other_user2.id}/')
```